SonarQube.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```diff
@@ -413,36 +413,54 @@
-  -->
+       
+       Conditions:
+       * project is a test project -> turn off Roslyn analysis
+       * project is excluded -> turn off Roslyn analysis
+       * downloaded SonarLint ruleset and analyzer found -> add SonarLint config to existing analysis config
+       * downloaded SonarLint ruleset and analyzer not found -> no change to existing config
+   -->
   <Target Name="OverrideRoslynCodeAnalysisProperties"
         Condition=" $(SonarQubeTempPath) != '' "
         DependsOnTargets="SonarQubeCategoriseProject"
         AfterTargets="ResolveCodeAnalysisRuleSet"
         BeforeTargets="CoreCompile">
 
+    <!-- Check whether to turn off code analysis -->
     <PropertyGroup>
-      <!-- Check if the ruleset exists -->
-      <SonarQubeRoslynRulesetFullName>$(SonarQubeConfigPath)SonarQubeRoslyn-cs.ruleset</SonarQubeRoslynRulesetFullName>
-      <SonarQubeRoslynRulesetExists>$([System.IO.File]::Exists($(SonarQubeRoslynRulesetFullName)))</SonarQubeRoslynRulesetExists>
-
-      <!-- Run FxCop only if the ruleset exists (else it means no Roslyn rules are enabled) -->
-      <SonarQubeRunRoslynCodeAnalysis>$(SonarQubeRoslynRulesetExists)</SonarQubeRunRoslynCodeAnalysis>
-
-      <!-- For the time being only run for C# -->
-      <SonarQubeRunRoslynCodeAnalysis Condition=" $(Language) != 'C#' ">false</SonarQubeRunRoslynCodeAnalysis>
-
-      <!-- Don't run FxCop if the project is excluded -->
-      <SonarQubeRunRoslynCodeAnalysis Condition=" $(SonarQubeExclude) == 'true' ">false</SonarQubeRunRoslynCodeAnalysis>
-
-      <!-- Don't run FxCop if the project is a test one -->
-      <SonarQubeRunRoslynCodeAnalysis Condition=" $(SonarQubeTestProject) == 'true' ">false</SonarQubeRunRoslynCodeAnalysis>
-
+      <SonarQubeDisableRoslynCodeAnalysis Condition="$(SonarQubeExclude) == 'true' OR $(SonarQubeTestProject) == 'true' ">true</SonarQubeDisableRoslynCodeAnalysis>
     </PropertyGroup>
-
-    <PropertyGroup Condition=" $(SonarQubeRunRoslynCodeAnalysis) != 'true' ">
+    
+    <PropertyGroup Condition=" $(SonarQubeDisableRoslynCodeAnalysis) == 'true' ">
       <ErrorLog></ErrorLog>
       <ResolvedCodeAnalysisRuleSet></ResolvedCodeAnalysisRuleSet>
+    </PropertyGroup>
+    <!-- else -->
+    <CallTarget Targets="SetRoslynCodeAnalysisProperties"
+                Condition="$(SonarQubeDisableRoslynCodeAnalysis) != 'true'"/>
+
+  </Target>
 
+  <Target Name="SetRoslynCodeAnalysisProperties">
+    
+    <PropertyGroup>
+      <!-- Check if the ruleset exists -->
+      <SonarQubeRoslynRulesetFullName>$(SonarQubeConfigPath)SonarQubeRoslyn-cs.ruleset</SonarQubeRoslynRulesetFullName>
+      <SonarQubeRoslynRulesetExists>$([System.IO.File]::Exists($(SonarQubeRoslynRulesetFullName)))</SonarQubeRoslynRulesetExists>
+
+      <!-- Check if the downloaded analyzer assembly exists -->
+      <SonarLintAnalyzerAssembly>$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), SonarLint.dll))</SonarLintAnalyzerAssembly>
```

Comment:
@dbolkensteyn yes, just appending the parent folder would be sufficient, but there isn't a convenient MSBuild function to return the parent folder  We could call the static Directory.GetParent(...) method, but the syntax is pretty ugly. We're also using GetDirectoryNameOfFilaAbove(...) to locate the assembly containing the tasks. It won't make much difference which method we use, but we should be consistent.